File: internal/core/entity/todo.go
package entity

import (
	"errors"
	"time"
)

type Todo struct {
	UserID    int64
	Name      string
	Content   string
	CreatedAt time.Time
}

func (t Todo) Validate() error {
	if t.UserID <= 0 {
		return errors.New("invalid user_id")
	}
	if len(t.Name) == 0 {
		return errors.New("name is empty")
	}
	return nil
}

File: src/main/java/com/uuhnaut69/app/todo/model/Todo.java
package com.uuhnaut69.app.todo.model;

import java.time.Instant;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.SequenceGenerator;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

/**
 * @author uuhnaut
 */
@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
public class Todo {

  @Id
  @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "todo_id_generator")
  @SequenceGenerator(name = "todo_id_generator", sequenceName = "todo_id_seq", allocationSize = 1)
  private Long id;

  private String task;

  private Long userId;

  @CreationTimestamp
  private Instant createdAt;

}
